### PR TITLE
Fixes Issue#103

### PR DESCRIPTION
**Pull request by mandarin6b0**

### CODE REVIEW

**Comment by mandarin6b0** [comment]
@pedantix is it good now?

File: Tests/RedisTests/RedisDataDecoderTests.swift [hunk 1]
```diff
@@ -136,6 +136,7 @@ class RedisDataDecoderTests: XCTestCase {
         let incompleteString = "*8\r\n$16\r\ntest8:1523640910\r\n$10\r\n1523640910\r\n$16\r\ntest9:15" +
         "23640913\r\n$10\r\n1523640913\r\n$17\r\ntest10:1523640916\r\n$10\r\n15"
         try bulkStringTestCase(protocolString: incompleteString, expectedString: nil)
+        try bulkStringTestCase(protocolString: "*2\r\n$5\r\ntest0\r\n", expectedString: nil)
```
[the remark this answers is not in the document]
Comment (mandarin6b0) [reply Tests/RedisTests/RedisDataDecoderTests.swift]: Sure, I'll move it. Should I add separate method for testing arrayParsing with nil, or bulkStringTestCase is good?

[the remark this answers is not in the document]
Comment (pedantix) [reply Tests/RedisTests/RedisDataDecoderTests.swift]: https://github.com/vapor/redis/blob/master/Tests/RedisTests/RedisDataDecoderTests.swift#L204

this is where I  put Array assertions.

[the remark this answers is not in the document]
Comment (mandarin6b0) [reply Tests/RedisTests/RedisDataDecoderTests.swift]: I mean should I use for testing on nil `bulkStringTestCase` or another method. Then I found `assertWillParseNil`. I will stick with it, ok? :)
